### DISCUSSION TITLE
[권준] assignment 기능구현

### DISCUSSION THREAD

**Discussion by jun-05** [Wanted-07-team-9/pre-onboarding-7th-2-1-9#10] [기능구현]
# 배포링크
[배포링크](http://wanted-pre-onboarding-fe-7-june.s3-website.ap-northeast-2.amazonaws.com)

## 사용 Tools
![React](https://img.shields.io/badge/react-%2320232a.svg?style=for-the-badge&logo=react&logoColor=%2361DAFB) ![Redux](https://img.shields.io/badge/redux-%23593d88.svg?style=for-the-badge&logo=redux&logoColor=white) ![TailwindCSS](https://img.shields.io/badge/tailwindcss-%2338B2AC.svg?style=for-the-badge&logo=tailwind-css&logoColor=white)

## Assignment 1:heavy_check_mark:
- 필수 요구 사항 
    - Figma 상의 디자인 및 기능 구현 
        - 차량 리스트 
            - 차량이 없을 때 처리
            - 차량 불러오는 중 처리
        - 차량 상세

#### redux 텅크를 사용하여 상태관리를 하였습니다.
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/0b67651e58ec37eaf129dedb953a7e03aa988399/src/modules/car/thunks.js#L18-L29

#### 상태관리로 carList 와 carWithType, 이 두개를 사용하였고, 조건에 따라 useState값을 사용하여, 컴포넌트에 전역state값을 전해주었습니다
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/f21894c3617c05eb69acd94b4fc70ad09b7a266f/src/container/main/CarContainer.jsx#L9-L43

#### API에서 받아오는 fuelType과 segment를 2차원 배열로 만들고, 중복값을 제거하고,ENUM값을 추가하였습니다.
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/0b67651e58ec37eaf129dedb953a7e03aa988399/src/utils/tag.js#L16-L25

이부분은 제가 과제를 잘못이해 하여 구현한것 같습니다.
결과 상태값 예시 :  [ ["fuelType","gasoline","가솔린"],["segment","C",소형"],]

#### 컴포넌트에서 로딩중이거나, 오브젝트가 비어있을경우 각각 처리해주었습니다.
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/0b67651e58ec37eaf129dedb953a7e03aa988399/src/component/main/CarList.jsx#L11-L15

#### 전역값을 사용하여 같은 태그를 눌러도 재로딩이 되지 않도록 하였고, 선택된 태그의 정보를 저장하여 뒤로가기 버튼 클릭시에도 값을 불러올수있도록 하였습니다. 
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/0b67651e58ec37eaf129dedb953a7e03aa988399/src/component/main/TagList.jsx#L10-L22

#### 

## - 추가 구현 사항 :heavy_multiplication_x:
   - SEO
        - 카카오톡, 페이스북에 공유 시 아래의 내용이 미리보기로 노출되도록 해야 함
            - 제목: car.brand + [car.name](http://car.name/)
            - 내용: 월 car.amount 원
            - 사진: 차량 사진

SEO는 구현하지 못했습니다.

**Comment by songseul** [comment]
- 필터 버튼을 클릭해도 재 렌더링 되지 않고 뒤로 가기 버튼을 눌러도 이전 값을 불러오도록 전역값을 사용 한 점이 인상 깊었습니다.
- 필터링 버튼 부분이 scroll 된다면 더 좋을것 같습니다 
- 디테일 페이지에 대한 부분은 어떻게 관리 되고 있는지 궁금 합니다

**Comment by im-binary** [comment]
- 선택된 태그의 정보를 저장하여 뒤로가기 버튼 클릭시에도 값을 불러올 수 있도록 한 점이 멋있네요
- 로딩이나 데이터가 없을 때 에러를 간결하게 잘 처리하신 것 같습니다.

**Comment by tirhande** [comment]
- Redux 사용하신점과 tailwinds css 적용하신게 좋아요!
- 태그 클릭시 데이터 재로딩 안하게 구현하신 부분이 인상 깊어요.
- JSON.stringify(carList)가 아닌 length 로 체크하셨으면 하는 조금 아쉬운 부분이 있어요!

**Comment by anotheranotherhoon** [comment]
- redux를 통해서 상태관리를 일목요연하게 하신게 인상적입니다!

**Comment by NR0617** [comment]
변수명이 의미를 잘 표현해서 좋았습니다 긴 변수명은 조금 줄일 수 있으면 좋겠습니다
컴포넌트에서 비어있을 경우 NotFound 컴포넌트를 보이도록 처리해준 부분이 좋았습니다!
